### PR TITLE
Add missing User endpoints for Management API

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

Add methods for missing Users endpoints on the Management API:

- `\Auth0\SDK\API\Management\Users::getRoles()`
- `\Auth0\SDK\API\Management\Users::removeRoles()`
- `\Auth0\SDK\API\Management\Users::addRoles()`
- `\Auth0\SDK\API\Management\Users::getEnrollments()`
- `\Auth0\SDK\API\Management\Users::getPermissions()`
- `\Auth0\SDK\API\Management\Users::removePermissions()`
- `\Auth0\SDK\API\Management\Users::addPermissions()`
- `\Auth0\SDK\API\Management\Users::getLogs()`
- `\Auth0\SDK\API\Management\Users::generateRecoveryCode()`
- `\Auth0\SDK\API\Management\Users::invalidateBrowsers()`

## References

[Management API Users docs](https://auth0.com/docs/api/management/v2#!/Users/get_users)

### Testing

- [x] This change adds test coverage

### Checklist

- [x] All existing and new tests complete without errors.
- [x] The correct base branch is being used.

### CODE REVIEW

File: src/API/Management/Users.php [hunk 1]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param array  $params  Additional listing params like page, per_page, and include_totals.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_user_roles
+     */
+    public function getRoles($user_id, array $params = [])
```
Comment (peter279k) [inline src/API/Management/Users.php]: How about adding type hint for `$user_id`?

```
.......
public function getRoles(string $user_id, array $params = [])
.......
```

File: src/API/Management/Users.php [hunk 2]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws EmptyOrInvalidParameterException Thrown if the roles parameter is empty.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/delete_user_roles
+     */
+    public function removeRoles($user_id, array $roles)
```
Comment (peter279k) [inline src/API/Management/Users.php]: How about adding type hint for `$user_id`?

File: src/API/Management/Users.php [hunk 3]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws EmptyOrInvalidParameterException Thrown if the roles parameter is empty.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/post_user_roles
+     */
+    public function addRoles($user_id, array $roles)
```
Comment (peter279k) [inline src/API/Management/Users.php]: How about adding type hint for `$user_id`?

File: src/API/Management/Users.php [hunk 4]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param string $user_id User ID to invalidate browsers for.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/post_invalidate_remember_browser
+     */
+    public function invalidateBrowsers($user_id)
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 5]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param string $user_id User ID to remove and generate recovery codes for.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/post_recovery_code_regeneration
+     */
+    public function generateRecoveryCode($user_id)
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 6]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param array  $params  Additional listing params like page, per_page, sort, and include_totals.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_logs_by_user
+     */
+    public function getLogs($user_id, array $params = [])
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 7]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws InvalidPermissionsArrayException Thrown if the permissions parameter is malformed.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/post_permissions
+     */
+    public function addPermissions($user_id, array $permissions)
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 8]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws InvalidPermissionsArrayException Thrown if the permissions parameter is malformed.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/delete_permissions
+     */
+    public function removePermissions($user_id, array $permissions)
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 9]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param array  $params  Additional listing params like page, per_page, and include_totals.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_permissions
+     */
+    public function getPermissions($user_id, array $params = [])
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

File: src/API/Management/Users.php [hunk 10]
```diff
@@ -242,6 +238,285 @@ public function deleteMultifactorProvider($user_id, $mfa_provider)
...
+     * @param string $user_id User ID to get enrollments for.
+     *
+     * @throws EmptyOrInvalidParameterException Thrown if the user_id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @return mixed
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_enrollments
+     */
+    public function getEnrollments($user_id)
```
Comment (peter279k) [inline src/API/Management/Users.php]: It's same issue I mention on previous comment.

**Comment by peter279k** [comment]
I also found that this PHP package still requires `php-5.5` version and it's not compatible with adding the `string` type hint for specific variable.

File: src/API/Management/GenericResource.php [hunk 11]
```diff
@@ -106,6 +106,7 @@ protected function checkInvalidPermissions(array $permissions)
         }
 
         foreach ($permissions as $permission) {
+
```
Comment (luisrudge) [inline src/API/Management/GenericResource.php]: typo?

**Comment by joshcanhelp** [comment]
@peter279k - Correct, we still support PHP 5.5 currently so `string` type hints are not allowed.

File: src/API/Management/GenericResource.php (hunk 11 above)
Comment (joshcanhelp) [reply src/API/Management/GenericResource.php]: It's iterating through an array, the `$permission` is the current array element.

**Pull request merged**